rollout/trafficrouting/istio/controller_test.go

```diff
@@ -78,6 +79,118 @@ func TestGetReferencedVirtualServices(t *testing.T) {
 	})
 }
 
+const istiovsvc1 = `apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: istio-vsvc1-name
+  namespace: default
+spec:
+  gateways:
+  - istio-rollout-gateway
+  hosts:
+  - istio-rollout.dev.argoproj.io
+  http:
+  - name: primary
+    route:
+    - destination:
+        host: 'stable'
+      weight: 100
+    - destination:
+        host: canary
+      weight: 0
+  - name: secondary
+    route:
+    - destination:
+        host: 'stable'
+      weight: 100
+    - destination:
+        host: canary
+      weight: 0`
+
+func TestGetReferencedMultipleVirtualServices(t *testing.T) {
+
+	multipleVirtualService := []v1alpha1.IstioVirtualService{{Name: "istio-vsvc1-name", Routes: nil}, {Name: "istio-vsvc2-name", Routes: nil}}
+
+	ro := v1alpha1.Rollout{
+		Spec: v1alpha1.RolloutSpec{
+			Strategy: v1alpha1.RolloutStrategy{
+				Canary: &v1alpha1.CanaryStrategy{},
+			},
+		},
+	}
+	ro.Spec.Strategy.Canary.TrafficRouting = &v1alpha1.RolloutTrafficRouting{
+		Istio: &v1alpha1.IstioTrafficRouting{
+			VirtualServices: multipleVirtualService,
+		},
+	}
+	ro.Namespace = metav1.NamespaceDefault
+
+	vService := unstructuredutil.StrToUnstructuredUnsafe(istiovsvc1)
+
+	t.Run("get referenced virtualService - fail", func(t *testing.T) {
+		c := NewFakeIstioController(vService)
+		_, err := c.GetReferencedVirtualServices(&ro)
+		expectedErr := field.Invalid(field.NewPath("spec", "strategy", "canary", "trafficRouting", "istio", "virtualServices", "name"), "istio-vsvc2-name", "virtualservices.networking.istio.io \"istio-vsvc2-name\" not found")
+		assert.Equal(t, expectedErr.Error(), err.Error())
+	})
+}
+
+func TestGetReferencedMultipleVirtualServicesInvalidConfig(t *testing.T) {
+
+	multipleVirtualService := []v1alpha1.IstioVirtualService{{Name: "istio-vsvc1-name", Routes: nil}}
+
+	ro := v1alpha1.Rollout{
+		Spec: v1alpha1.RolloutSpec{
+			Strategy: v1alpha1.RolloutStrategy{
+				Canary: &v1alpha1.CanaryStrategy{},
+			},
+		},
+	}
+	ro.Spec.Strategy.Canary.TrafficRouting = &v1alpha1.RolloutTrafficRouting{
+		Istio: &v1alpha1.IstioTrafficRouting{
+			VirtualService: v1alpha1.IstioVirtualService{
+				Name: "istio-vsvc1-name",
+			},
+			VirtualServices: multipleVirtualService,
+		},
+	}
+	ro.Namespace = metav1.NamespaceDefault
+
+	vService := unstructuredutil.StrToUnstructuredUnsafe(istiovsvc1)
+
+	t.Run("get referenced virtualService - fail", func(t *testing.T) {
+		c := NewFakeIstioController(vService)
+		_, err := c.GetReferencedVirtualServices(&ro)
+		fldPath := field.NewPath("spec", "strategy", "canary", "trafficRouting", "istio")
+		expected := fmt.Sprintf("%s: Internal error: either VirtualService or VirtualServices must be configured", fldPath)
+		assert.Equal(t, expected, err.Error())
+	})
+}
+
+func TestGetReferencedNoVirtualServicesConfig(t *testing.T) {
```

Comment:
This case can be added to `TestGetReferencedMultipleVirtualServicesInvalidConfig` to reduce some duplication.